code/RedirectedURLCsvBulkLoader.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+class RedirectedURLCsvBulkLoader extends CsvBulkLoader
+{
+    private $exception_thrown;
+
+    /**
+     * @todo Better messages for relation checks and duplicate detection
+     * Note that columnMap isn't used.
+     * overriding this here for handling ValidationException when DataObject::write() called
+     *
+     * @param array $record
+     * @param array $columnMap
+     * @param BulkLoader_Result $results
+     * @param boolean $preview
+     *
+     * @return int
+     */
+    protected function processRecord($record, $columnMap, &$results, $preview = false) {
+        // make $exception_thrown false initially
+        $this->exception_thrown = false;
```

Comment:
Why are we storing this against the object instance instead of just as a local variable? I think `$exceptionThrown` is fine as a local variable.